diff --git a/drivers/usb/dwc3/core.h b/drivers/usb/dwc3/core.h
index a6cf7c07b913..dc5fb87497ab 100644
--- a/drivers/usb/dwc3/core.h
+++ b/drivers/usb/dwc3/core.h
@@ -1617,6 +1617,8 @@ int dwc3_send_gadget_generic_command(struct dwc3 *dwc, unsigned int cmd,
 void dwc3_gadget_clear_tx_fifos(struct dwc3 *dwc);
 void dwc3_remove_requests(struct dwc3 *dwc, struct dwc3_ep *dep, int status);
 int dwc3_core_init(struct dwc3 *dwc);
+void dwc3_gadget_enter_hibernation(struct dwc3 *dwc);
+void dwc3_gadget_exit_hibernation(void *_dwc);
 #else
 static inline int dwc3_gadget_init(struct dwc3 *dwc)
 { return 0; }
@@ -1638,6 +1640,10 @@ static inline int dwc3_send_gadget_generic_command(struct dwc3 *dwc,
 { return 0; }
 static inline void dwc3_gadget_clear_tx_fifos(struct dwc3 *dwc)
 { }
+static inline void dwc3_gadget_enter_hibernation(struct dwc3 *dwc)
+{ }
+static inline void dwc3_gadget_exit_hibernation(void *_dwc)
+{ }
 #endif
 
 #if IS_ENABLED(CONFIG_USB_DWC3_OTG) || IS_ENABLED(CONFIG_USB_DWC3_DUAL_ROLE)
diff --git a/drivers/usb/dwc3/gadget.h b/drivers/usb/dwc3/gadget.h
index 7353905ca7d2..03c255ae61ea 100644
--- a/drivers/usb/dwc3/gadget.h
+++ b/drivers/usb/dwc3/gadget.h
@@ -135,8 +135,6 @@ void dwc3_stop_active_transfer(struct dwc3_ep *dep, bool force,
 		bool interrupt);
 int dwc3_gadget_run_stop(struct dwc3 *dwc, int is_on, int suspend);
 dma_addr_t dwc3_trb_dma_offset(struct dwc3_ep *dep, struct dwc3_trb *trb);
-void dwc3_gadget_enter_hibernation(struct dwc3 *dwc);
-void dwc3_gadget_exit_hibernation(void *_dwc);
 void dwc3_ep0_send_delayed_status(struct dwc3 *dwc);
 
 /**
diff --git a/drivers/usb/dwc3/host.c b/drivers/usb/dwc3/host.c
index 080bfe964f3a..d5eda659344d 100644
--- a/drivers/usb/dwc3/host.c
+++ b/drivers/usb/dwc3/host.c
@@ -20,6 +20,7 @@ void dwc3_host_wakeup_register(dwc3_wakeup_t func)
 {
 	dwc3_wakeup_fn = func;
 }
+EXPORT_SYMBOL_GPL(dwc3_host_wakeup_register);
 
 /* callback function */
 void dwc3_host_wakeup_capable(struct device *dev, bool wakeup)
@@ -27,6 +28,7 @@ void dwc3_host_wakeup_capable(struct device *dev, bool wakeup)
 	if (dwc3_wakeup_fn)
 		dwc3_wakeup_fn(dev, wakeup);
 }
+EXPORT_SYMBOL_GPL(dwc3_host_wakeup_capable);
 
 static void dwc3_host_fill_xhci_irq_res(struct dwc3 *dwc,
 					int irq, char *name)
